dotnet/src/Connectors/Connectors.OpenAIV2.UnitTests/Services/OpenAIFileServiceTests.cs

```diff
@@ -0,0 +1,298 @@
+        {
+            var file = await service.GetFileAsync("file-id");
+            Assert.NotNull(file);
+            Assert.NotEqual(string.Empty, file.Id);
+            Assert.NotEqual(string.Empty, file.FileName);
+            Assert.NotEqual(DateTime.MinValue, file.CreatedTimestamp);
+            Assert.NotEqual(0, file.SizeInBytes);
+        }
+    }
+
+    [Theory]
+    [InlineData(true, true)]
+    [InlineData(false, true)]
+    [InlineData(true, false)]
+    [InlineData(false, false)]
+    public async Task GetFilesWorksCorrectlyAsync(bool isAzure, bool isFailedRequest)
+    {
+        // Arrange
+        var service = this.CreateFileService(isAzure);
+        using var response =
+            isFailedRequest ?
+                this.CreateFailedResponse() :
+                this.CreateSuccessResponse(
+                    """
+                    {
+                        "data": [
+                            {
+                                "id": "123",
+                                "filename": "file1.txt",
+                                "purpose": "assistants",
+                                "bytes": 120000,
+                                "created_at": 1677610602
+                            },
+                            {
+                                "id": "456",
+                                "filename": "file2.txt",
+                                "purpose": "assistants",
+                                "bytes": 999,
+                                "created_at": 1677610606
+                            }
+                        ]
+                    }
+                    """);
+        this._messageHandlerStub.ResponseToReturn = response;
+
+        // Act & Assert
+        if (isFailedRequest)
+        {
+            await Assert.ThrowsAsync<HttpOperationException>(() => service.GetFilesAsync());
+        }
+        else
+        {
+            var files = (await service.GetFilesAsync()).ToArray();
+            Assert.NotNull(files);
+            Assert.NotEmpty(files);
+        }
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task GetFileContentWorksCorrectlyAsync(bool isAzure)
+    {
+        // Arrange
+        var data = BinaryData.FromString("Hello AI!");
+        var service = this.CreateFileService(isAzure);
+        this._messageHandlerStub.ResponseToReturn =
+            new HttpResponseMessage(System.Net.HttpStatusCode.OK)
+            {
+                Content = new ByteArrayContent(data.ToArray())
+            };
+
+        // Act & Assert
+        var content = await service.GetFileContentAsync("file-id");
+        var result = content.Data!.Value;
+        Assert.Equal(data.ToArray(), result.ToArray());
+    }
+
+    [Theory]
+    [InlineData(true, true)]
+    [InlineData(false, true)]
+    [InlineData(true, false)]
+    [InlineData(false, false)]
+    public async Task UploadContentWorksCorrectlyAsync(bool isAzure, bool isFailedRequest)
```

Comment:
Consider splitting this and others into two tests one for success and one for failed. Not necessary but I think it will be cleaner.